File: clients/php/src/AlreadyExistsException.php
<?php

declare(strict_types=1);

namespace FileDBv2;

/**
 * Thrown when a keyed insert collides with a key already held by a live record —
 * the client mapping of the gRPC ALREADY_EXISTS status. Raised by insert() when
 * a `key` is supplied and that key is already taken. Use upsert() for
 * insert-or-replace semantics instead.
 */
class AlreadyExistsException extends FileDBException
{
}

File: clients/java/src/main/java/com/srjn45/filedbv2/AlreadyExistsException.java
package com.srjn45.filedbv2;

/**
 * A keyed insert used a key already held by a live record.
 *
 * <p>Raised from {@link FileDBClient#insert(String, java.util.Map, long, String)}
 * (and its keyed overloads) when the engine answers {@code ALREADY_EXISTS}.
 */
public class AlreadyExistsException extends FileDBException {

    public AlreadyExistsException(String message, Throwable cause) {
        super(message, cause);
    }
}
